File: ipc_shared_memory.py
import sysv_ipc
import os
import time
import warnings # Import the warnings module
# Ignore warnings from sysv_ipc library
with warnings.catch_warnings():
warnings.simplefilter("ignore")
key = sysv_ipc.ftok("shmfile", 65)
shm = sysv_ipc.SharedMemory(key, sysv_ipc.IPC_CREAT, 0o666, 1024)
shm.write("Hello from shared memory!")
time.sleep(20) # Wait for 20 seconds before the terminated
shm.detach()
shm.remove()

File: ipc_shared_memory.c
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <sys/ipc.h>
#include <sys/shm.h>
#include <sys/types.h>
#include <unistd.h>
#define SHM_SIZE 1024
int main() {
int shmid;
key_t key;
char *shm, *s;
key = ftok("shmfile", 65);
shmid = shmget(key, SHM_SIZE, 0666 | IPC_CREAT);
shm = shmat(shmid, NULL, 0);
memcpy(shm, "Hello from shared memory!", 25);
sleep(20); // sleep 20 seconds before the message is terminated
shmdt(shm);
shmctl(shmid, IPC_RMID, NULL);
return 0;
}
